linera-core/src/unit_tests/test_utils.rs

```diff
@@ -333,7 +333,9 @@ where
         blobs: Vec<Blob>,
     ) -> Option<Result<ChainInfoResponse, NodeError>> {
         match validator.fault_type {
-            FaultType::DontProcessValidated if T::IS_VALIDATED => None,
+            FaultType::DontProcessValidated if matches!(T::KIND, CertificateKind::Validated) => {
```

Comment:
(Since it's a simple enum I'd just implement `Eq` for it and use `T::KIND == CertificateKind::Validated` here.)